src/Surfnet/ServiceProviderDashboard/Application/Metadata/OidcngJsonGenerator.php

```diff
@@ -49,13 +50,22 @@ public function __construct(
     ) {
     }
 
-    public function generateForNewEntity(ManageEntity $entity, string $workflowState): array
-    {
+    public function generateForNewEntity(
```

Comment:
Looks like a lot of duplication here

Reply:
But still adheres to the dry principles. As every generator is creating a json payload for a different Manage entity type. Each having characteristics of their own. 

You are onto something good here tho. Much business logic is now present in these generators (which are very low level) and that knowledge should be in the model somewhere. Once that distinction is made, the generators will become much more generic. And we can re-evaluate if they can be merged into one.